ssl/ssl_lib.c

```diff
@@ -5401,6 +5401,38 @@ int SSL_client_hello_get1_extensions_present(SSL *s, int **out, size_t *outlen)
     return 0;
 }
 
+int SSL_client_hello_get_raw_extensions(SSL *s, unsigned short *out, size_t *outlen)
+{
+    RAW_EXTENSION *ext;
+    size_t num = 0, i;
+
+    if (s->clienthello == NULL || outlen == NULL)
+        return 0;
+    for (i = 0; i < s->clienthello->pre_proc_exts_len; i++) {
+        ext = s->clienthello->pre_proc_exts + i;
+        if (ext->present)
+            num++;
+    }
+    if (num == 0) {
+        *outlen = 0;
+        return 1;
+    }
+    if (out == NULL) {
+        *outlen = num;
```

Comment:
The way you did it, the parameter should be named `*num_exts` or in some similar way. I'd also like to see it to be input/output parameter so in case someone preallocates the buffer to hold some number of extensions it can be checked. I.E., `if (*num_exts < num) return 0;` And perhaps raise some error.

Reply:
Updated for `*num_exts`

For `raise some error`, if it has to, could you please suggest a proper way? thanks.